src/ejabberd_config.erl

```diff
@@ -1127,9 +1127,29 @@ compute_config_version(LC, LCH) ->
     crypto:hash(sha, term_to_binary(L1)).
 
 compute_config_file_version(#state{opts = Opts, hosts = Hosts}) ->
-    L = sort_config(Opts ++ Hosts),
+    Opts2 = filter_out_gd_endpoints(Opts),
+    L = sort_config(Opts2 ++ Hosts),
     crypto:hash(sha, term_to_binary(L)).
 
+filter_out_gd_endpoints([]) ->
+    [];
+filter_out_gd_endpoints([{local_config, {modules, Host}, Mods} | Opts]) ->
+    NewMods = delete_path_in_proplist(Mods, [mod_global_distrib, connections, endpoints]),
+    [{local_config, {modules, Host}, NewMods} | Opts];
+filter_out_gd_endpoints([Opt | Opts]) ->
+    [Opt | filter_out_gd_endpoints(Opts)].
+
+delete_path_in_proplist(Plist, [Step]) ->
```

Comment:
@fenek, if you know of any function in stdlib that could replace this definition (that is a solution for a very generic problem of nested proplists), let me know - I haven't found one.